server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/StateChange.scala

```diff
@@ -44,10 +41,7 @@ case object MailboxTypeName extends TypeName {
     case _ => None
   }
 
-  override def parseState(string: String): Either[IllegalArgumentException, UuidState] = Try(UUID.fromString(string))
-    .toEither
-    .map(UuidState(_))
-    .left.map(new IllegalArgumentException(_))
+  override def parseState(string: String): Either[IllegalArgumentException, UuidState] = UuidState.parse(string)
```

Comment:
You override this method with the same syntax in every of your subclasses... Why not defining this only once as a default method instead in your `TypeName` trait?

Reply:
That is why I am doing this PR: to make TypeState(CustomTypeName -> CustomState) customization is able.
Every state now should not tied to Uuid State, cause we have some custom State like FilterState(which is parsed from Integer).

Reply:
Well if you need to customize it you can override it then, I mean it's not because you have a default method that you can't override it if you need to?

Reply:
Ok i think i get your point now. It's a nice improved suggestion for me :)